### PR TITLE
Display the node sources as a table grid

### PR DESCRIPTION
**Pull request by oanaBiancaSchiopu**

### CODE REVIEW

**Review by tao-xinxiu — APPROVED** [review]

File: rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/TreeView.java [hunk 1]
```diff
@@ -103,11 +120,59 @@ private class TNS extends TreeNode {
...
+
+        NodeSourceDisplayedDescription(String description) {
+
+            infrastructure = beautifyName(description.substring(description.indexOf(" ") + 1,
+                                                                description.indexOf(POLICY)));
+            if (infrastructure.contains(",")) {
+                infrastructure = infrastructure.substring(0, infrastructure.indexOf(","));
+            }
+            if (infrastructure.contains(INFRASTRUCTURE_FIELD)) {
+                infrastructure = infrastructure.substring(0, infrastructure.indexOf(INFRASTRUCTURE_FIELD));
```
Comment (fviale) [inline rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/TreeView.java:149]: why not use infrastructure = infrastructure.replace(INFRASTRUCTURE_FIELD, "")

File: rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/TreeView.java [hunk 2]
```diff
@@ -103,11 +120,59 @@ private class TNS extends TreeNode {
...
+            }
+            if (infrastructure.contains("Manager")) {
+                infrastructure = infrastructure.substring(0, infrastructure.indexOf("Manager"));
```
Comment (fviale) [inline rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/TreeView.java:152]: same remark

File: rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/TreeView.java [hunk 3]
```diff
@@ -103,11 +120,59 @@ private class TNS extends TreeNode {
...
     }
 
+    private class NodeSourceDisplayedDescription {
+        private String infrastructure;
+
+        private String policy;
+
+        private String access;
+
+        NodeSourceDisplayedDescription(String description) {
```
Comment (fviale) [inline rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/TreeView.java:141]: I would put the code that split the description in different categories first and then apply the replacements/beautify. The code will be easier to read

File: rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/TreeView.java [hunk 4]
```diff
@@ -103,11 +120,59 @@ private class TNS extends TreeNode {
...
+
+            infrastructure = beautifyName(description.substring(description.indexOf(" ") + 1,
+                                                                description.indexOf(POLICY)));
+            if (infrastructure.contains(",")) {
+                infrastructure = infrastructure.substring(0, infrastructure.indexOf(","));
+            }
+            if (infrastructure.contains(INFRASTRUCTURE_FIELD)) {
```
Comment (fviale) [inline rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/TreeView.java:148]: this replacement can be done in a single line:

infrastructure = infrastructure.replace(INFRASTRUCTURE_FIELD, "")

Comment (fviale) [reply rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/TreeView.java:148]: I mean, you do the replacements after you do the initial split. e.g.:

```
        String infrastructure = nodeSourceDescription.split(" ")[1];
        if (infrastructure.contains(",")) {
            infrastructure = infrastructure.substring(0, infrastructure.indexOf(","));
        }
```

File: rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/TreeView.java [hunk 5]
```diff
@@ -103,11 +120,55 @@ private class TNS extends TreeNode {
...
+
+            if (infrastructure.contains(",")) {
+                infrastructure = infrastructure.substring(0, infrastructure.indexOf(","));
+            }
+            infrastructure = infrastructure.replace(INFRASTRUCTURE_FIELD, "");
+            infrastructure = infrastructure.replace("Manager", "");
+            infrastructure = beautifyName(infrastructure);
+
+            policy = policy.substring(0, policy.indexOf(USER_ACCESS_TYPE_));
+            policy = policy.replace(POLICY_FIELD, "");
```
Comment (fviale) [inline rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/TreeView.java:155]: could you call beautifyName on the policy also ?

**Review by fviale — APPROVED** [review]
LGTM

**Pull request merged**